assembly/std/storage.ts

```diff
@@ -188,49 +280,72 @@ export function appendOf<T>(address: Address, key: T, value: T): void {
 }
 
 /**
- * Checks if the (key, value) exists in the datastore
- * of the callee's address.
+ * Checks if the key-value pair exists in the datastore of the current contract.
+ *
+ * @typeParam T - Can be either `string`, `Args`, or `StaticArray<u8>`.
+ *
+ * @param key - The key to check for existence in the datastore.
+ *
+ * @returns A boolean value indicating whether the key-value pair exists in the datastore or not.
  *
- * @typeParam T - `string`, `Args` or `StaticArray<u8>`
- * @param key -
+ * @throws
+ * - [compilation only] if the `key` type are neither `string`, `StaticArray<u8>`, or `Args`.
  */
 export function has<T>(key: T): bool {
   return env.has(toDatastoreFormat(key));
 }
 
 /**
- * Checks if the (key, value) exists in the datastore
- * of the given address.
+ * Checks if the key-value pair exists in the datastore of the target address.
  *
- * @typeParam T - `string`, `Args` or `StaticArray<u8>`
- * @param address -
- * @param key -
+ * @remarks Can only be called at smart contract generation time by the parent smart contract to the child's address.
  *
+ * @typeParam T - Can be either `string`, `Args`, or `StaticArray<u8>`.
+ *
+ * @param address - The child smart contract `address`.
+ * @param key - The key to check for existence in the datastore.
+ *
+ * @returns A boolean value indicating whether the key-value pair exists in the datastore or not.
+ *
+ * @throws
+ * - if the contract at the given `address` does not exist
+ * - if the caller is not authorized to modify the datastore.
+ * - [compilation only] if the `key` type are neither `string`, `StaticArray<u8>`, or `Args`.
  */
 export function hasOf<T>(address: Address, key: T): bool {
   return env.hasOf(address.toString(), toDatastoreFormat(key));
 }
 
 /**
- *  Sets the executable bytecode of the callee's address.
+ * Sets the executable bytecode of the current address.
  *
- * @privateRemarks
- * TODO: explains failure consequences.
+ * @param bytecode - The bytecode to be set. It should be a `StaticArray<u8>`.
  *
- * @param bytecode -
+ * @throws
+ * - if the caller address does not correspond to a smart contract in the ledger.
+ * - if the bytecode size exceeds maximum allowed size.
+ * - if the address does not exist
+ * - if the caller lacks write permissions on the address.
```

Comment:
`Sets the executable bytecode of the current address`

As I understand, there is no address parameter involved, it will take Context.caller()

```suggestion
```

Reply:
we had a discussion about this here: https://github.com/massalabs/massa-as-sdk/pull/189#discussion_r1157043161